manifests/director.pp

```diff
@@ -71,4 +84,18 @@
       tag         => ['bareos', 'bareos_director'],
     }
   }
+
+  create_resources(::bareos::director::catalog, $catalogs)
```

Comment:
Instead of using the legacy create_resources, please iterate on the hashes as shown in the documentation: 
https://puppet.com/docs/puppet/4.10/lang_resources_advanced.html#implementing-the-createresources-function